packages/ethereum/contracts/HoprChannels.sol

```diff
@@ -298,35 +297,24 @@ contract HoprChannels is IERC777Recipient, ERC1820Implementer {
         bytes calldata operatorData
     ) external override {
         require(msg.sender == address(token), "caller must be HoprToken");
-
-        if (
-            operator == address(this) || // must not be triggered by HoprChannels
-            from == address(0) // ignore 'mint'
-        ) {
-            return;
-        }
+        require(to == address(this), "must be sending tokens to HoprChannels");
 
         // must be one of our supported functions
-        require(
-            userData.length == FUND_CHANNEL_MULTI_SIZE,
-            "userData must match one of our supported functions"
-        );
-
-        address account1;
-        address account2;
-        uint256 amount1;
-        uint256 amount2;
-
-        (account1, account2, amount1, amount2) = abi.decode(userData, (address, address, uint256, uint256));
-        require(amount == amount1.add(amount2), "amount sent must be equal to amount specified");
-
-        //require(from == account1 || from == account2, "funder must be either account1 or account2");
-
-        if (amount1 > 0){
-          _fundChannel(account1, account2, amount1);
-        }
-        if (amount2 > 0){
-          _fundChannel(account2, account1, amount2);
+        if (userData.length == FUND_CHANNEL_MULTI_SIZE) {
```

Comment:
Why not fail if users try to call inappropriate function(s)?

Reply:
if the userData doesn't match we dont do anything, this is needed because HoprChannel uses transferFrom to pull funds in a different method

Reply:
> if the userData doesn't match we dont do anything

I understand

> this is needed because HoprChannel uses transferFrom to pull funds in a different method

Could you give an example?

Reply:
https://github.com/hoprnet/hoprnet/blob/641a43a7d89499e2c09fb500ae04ed0534506c38/packages/ethereum/contracts/HoprChannels.sol#L119

Reply:
Thx!